### ISSUE TITLE
node-gyp rebuild failing on "npm install --save hubot-hipchat" step

### ISSUE THREAD

**Issue by jeffelrod** [hipchat/hubot-hipchat#199]
OSX: Version 10.9.5 (13F34)
```node -v```: v.0.10.32
```npm -v```: 1.4.28
```hubot -v```: 2.7.5
```python --version```: Python 2.7.5

Have followed the instructions in README.md with outputs linked for each step:
$ [brew install icu4c](https://gist.github.com/jeffelrod/04d6022d1d8846b53c67#file-brew-install-icu4c-output)
$ [brew link icu4c --force](https://gist.github.com/jeffelrod/04d6022d1d8846b53c67#file-brew-link-icu4c-force-output)
$ [sudo npm install --global coffee-script hubot@v2.7.5](https://gist.github.com/jeffelrod/04d6022d1d8846b53c67#file-sudo-npm-install-global-coffee-script-hubot-v2-7-5-output)
$ [hubot --create ~/Documents/usefulbot](https://gist.github.com/jeffelrod/04d6022d1d8846b53c67#file-hubot-create-directory-output)
$ cd ~/Documents/usefulbot
$ [npm install](https://gist.github.com/jeffelrod/04d6022d1d8846b53c67#file-npm-install-output)

However, when I run ```npm install --save hubot-hipchat```, I get a long output with a bunch of error messages:
```
> node-stringprep@0.1.12 install /Users/jelrod/Documents/usefulbot/node_modules/hubot-hipchat/node_modules/node-xmpp/node_modules/node-xmpp-component/node_modules/node-xmpp-core/node_modules/node-stringprep
> node-gyp rebuild

  CXX(target) Release/obj.target/node_stringprep/node-stringprep.o
../node-stringprep.cc:2:10: fatal error: 'unicode/unistr.h' file not found
#include <unicode/unistr.h>
         ^
1 error generated.
make: *** [Release/obj.target/node_stringprep/node-stringprep.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:810:12)
gyp ERR! System Darwin 13.4.0
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/jelrod/Documents/usefulbot/node_modules/hubot-hipchat/node_modules/node-xmpp/node_modules/node-xmpp-component/node_modules/node-xmpp-core/node_modules/node-stringprep
gyp ERR! node -v v0.10.32
gyp ERR! node-gyp -v v1.0.1
gyp ERR! not ok 
npm WARN optional dep failed, continuing node-stringprep@0.1.12
 
> node-stringprep@0.2.3 install /Users/jelrod/Documents/usefulbot/node_modules/hubot-hipchat/node_modules/node-xmpp/node_modules/node-xmpp-core/node_modules/node-stringprep
> node-gyp rebuild

  CXX(target) Release/obj.target/node_stringprep/node-stringprep.o
../node-stringprep.cc:2:10: fatal error: 'unicode/unistr.h' file not found
#include <unicode/unistr.h>
         ^
1 error generated.
make: *** [Release/obj.target/node_stringprep/node-stringprep.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:810:12)
gyp ERR! System Darwin 13.4.0
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/jelrod/Documents/usefulbot/node_modules/hubot-hipchat/node_modules/node-xmpp/node_modules/node-xmpp-core/node_modules/node-stringprep
gyp ERR! node -v v0.10.32
gyp ERR! node-gyp -v v1.0.1
gyp ERR! not ok 
 
> node-expat@2.3.1 install /Users/jelrod/Documents/usefulbot/node_modules/hubot-hipchat/node_modules/node-xmpp/node_modules/node-expat
> node-gyp rebuild

  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmlparse.o
  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmltok.o
../deps/libexpat/lib/xmltok.c:471:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:484:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:504:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:517:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:730:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:749:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:762:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:775:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:871:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:890:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
10 warnings generated.
  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmlrole.o
  LIBTOOL-STATIC Release/libexpat.a
  CXX(target) Release/obj.target/node_expat/node-expat.o
  SOLINK_MODULE(target) Release/node_expat.node
  SOLINK_MODULE(target) Release/node_expat.node: Finished

> node-expat@2.1.4 install /Users/jelrod/Documents/usefulbot/node_modules/hubot-hipchat/node_modules/node-xmpp/node_modules/node-xmpp-component/node_modules/ltx/node_modules/node-expat
> node-gyp rebuild

  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmlparse.o
  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmltok.o
../deps/libexpat/lib/xmltok.c:471:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:484:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:504:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:517:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:730:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:749:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:762:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:775:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:871:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:890:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
10 warnings generated.
  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmlrole.o
  LIBTOOL-STATIC Release/libexpat.a
  CXX(target) Release/obj.target/node_expat/node-expat.o
  SOLINK_MODULE(target) Release/node_expat.node
  SOLINK_MODULE(target) Release/node_expat.node: Finished

> node-expat@2.1.4 install /Users/jelrod/Documents/usefulbot/node_modules/hubot-hipchat/node_modules/node-xmpp/node_modules/node-xmpp-client/node_modules/ltx/node_modules/node-expat
> node-gyp rebuild

  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmlparse.o
  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmltok.o
../deps/libexpat/lib/xmltok.c:471:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:484:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:504:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:517:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:730:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:749:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:762:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:775:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:871:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:890:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
10 warnings generated.
  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmlrole.o
  LIBTOOL-STATIC Release/libexpat.a
  CXX(target) Release/obj.target/node_expat/node-expat.o
  SOLINK_MODULE(target) Release/node_expat.node
  SOLINK_MODULE(target) Release/node_expat.node: Finished

> node-expat@2.1.4 install /Users/jelrod/Documents/usefulbot/node_modules/hubot-hipchat/node_modules/node-xmpp/node_modules/node-xmpp-server/node_modules/ltx/node_modules/node-expat
> node-gyp rebuild

  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmlparse.o
  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmltok.o
../deps/libexpat/lib/xmltok.c:471:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:484:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:504:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:517:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:730:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:749:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:762:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:775:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:871:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:890:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
10 warnings generated.
  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmlrole.o
  LIBTOOL-STATIC Release/libexpat.a
  CXX(target) Release/obj.target/node_expat/node-expat.o
  SOLINK_MODULE(target) Release/node_expat.node
  SOLINK_MODULE(target) Release/node_expat.node: Finished

> node-expat@2.1.4 install /Users/jelrod/Documents/usefulbot/node_modules/hubot-hipchat/node_modules/node-xmpp/node_modules/node-xmpp-core/node_modules/ltx/node_modules/node-expat
> node-gyp rebuild

  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmlparse.o
  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmltok.o
../deps/libexpat/lib/xmltok.c:471:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:484:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:504:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:517:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:730:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:749:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:762:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:775:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:871:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
../deps/libexpat/lib/xmltok.c:890:1: warning: missing field 'isName2'
      initializer [-Wmissing-field-initializers]
};
^
10 warnings generated.
  CC(target) Release/obj.target/expat/deps/libexpat/lib/xmlrole.o
  LIBTOOL-STATIC Release/libexpat.a
  CXX(target) Release/obj.target/node_expat/node-expat.o
  SOLINK_MODULE(target) Release/node_expat.node
  SOLINK_MODULE(target) Release/node_expat.node: Finished
npm ERR! node-stringprep@0.2.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-stringprep@0.2.3 install script.
npm ERR! This is most likely a problem with the node-stringprep package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls node-stringprep
npm ERR! There is likely additional logging output above.

npm ERR! System Darwin 13.4.0
npm ERR! command "node" "/usr/local/bin/npm" "install" "--save" "hubot-hipchat"
npm ERR! cwd /Users/jelrod/Documents/usefulbot
npm ERR! node -v v0.10.32
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR! not ok code 0
```

I'm not sure how to resolve any of the above - could someone point me in the right direction? Checked out https://github.com/hipchat/hubot-hipchat/issues/197, but the install isn't coming up successful for me in the end as it was for her, so I'm unsure how to proceed.

**Comment by tfhartmann** [comment]
+1

**Comment by atljeremy** [comment]
+1

**Comment by erinnovations** [comment]
+1 same on Ubuntu 14.10

**Comment by supercaracal** [comment]
:+1:

**Comment by crhatton** [comment]
Specifically, these 3 steps:
```
brew install icu4c
ln -s /usr/local/Cellar/icu4c/<VERSION>/bin/icu-config /usr/local/bin/icu-config
ln -s /usr/local/Cellar/icu4c/<VERSION>/include/* /usr/local/include
```

**Comment by NikolaMorena** [comment]
For me, hubot-hipchat installation fails when it tries to install node-expat 2.1.4..
I tried manual install of node-expat:
npm i node-expat
(works fine, installs latest version 2.3.7, but hubot-hipchat wants 2.1.4)
npm i node-expat@2.1.4 
Fails on both Ubuntu 14.04 and 12.04. Any ideas?

gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/morena/.nvm/versions/node/v0.12.2/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
gyp ERR! System Linux 3.13.0-40-generic
gyp ERR! command "node" "/home/morena/.nvm/versions/node/v0.12.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/morena/myhubot/node_modules/node-expat
gyp ERR! node -v v0.12.2
gyp ERR! node-gyp -v v1.0.3
gyp ERR! not ok
npm ERR! Linux 3.13.0-40-generic
npm ERR! argv "/home/morena/.nvm/versions/node/v0.12.2/bin/node" "/home/morena/.nvm/versions/node/v0.12.2/bin/npm" "i" "node-expat@2.1.4"
npm ERR! node v0.12.2
npm ERR! npm  v2.7.4
npm ERR! code ELIFECYCLE

npm ERR! node-expat@2.1.4 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!

**Comment by rbergman** [comment]
Several PRs have been merged and published, and builds on (at least) both OSX 10.10 and Ubuntu 14.04 should be working fine with Node 0.10.x or later.  README has been updated to remove the icu4c dependency.
